torch/_subclasses/fake_tensor.py

```diff
@@ -1732,23 +1739,19 @@ def validate(x):
             flat_arg_fake_tensors.append(x)
             return x
 
-        args, kwargs = tree_map_only(
-            torch.Tensor,
-            validate,
-            (args, kwargs),
-        )
-        return args, kwargs, flat_arg_fake_tensors
+        validated_args = [validate(a) for a in flat_args]
+        return validated_args, flat_arg_fake_tensors
 
-    def wrap_meta_outputs_with_default_device_logic(self, r, func, args, kwargs):
-        wrap = self.gen_wrap_fn(func, args, kwargs)
+    def wrap_meta_outputs_with_default_device_logic(self, r, func, flat_args, kwargs):
```

Comment:
nit. perhaps just pass `device rather than passing flat_args *and* kwargs. I found this a bit confusing.